### PR TITLE
Fix internal requests when gitea listens to unix socket or only external IP

### LINKED ISSUE #2232: UpdatePublicKey: unsupported protocol scheme "unix" when pushing via ssh
- Gitea version (or commit ref): https://github.com/go-gitea/gitea/commit/49df677c475d6a20575b99b5af8323f65937dadb
- Git version: 2.13.3
- Operating system: Linux 4.9.36

## Description

When the web server of gitea is configured to listen on a unix socket, users are unable to push changes via git ssh because the remote rejects with this error:

````
...
Writing objects: 100% (6/6), 2.13 KiB | 0 bytes/s, done.
Total 6 (delta 3), reused 0 (delta 0)
Gitea: Internal error
UpdatePublicKey: Post unix:///tmp/gitea:3000/api/internal/ssh/12/update: unsupported protocol scheme "unix"
...
 ! [remote rejected] master -> master (pre-receive hook declined)
````

Note that there are two issues there:

- `:3000` should obviously be not there on a unix socket
- The actual `Post` error. I think this is because Macaron does not support this scheme.

My `server` configuration:

````ini
[server]
PROTOCOL = unix
HTTP_ADDR = /tmp/gitea
HTTP_PORT =
````

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #2232

### CODE REVIEW

File: modules/private/branch.go [hunk 1]
```diff
@@ -20,9 +19,7 @@ func GetProtectedBranchBy(repoID int64, branchName string) (*models.ProtectedBra
 	reqURL := setting.LocalURL + fmt.Sprintf("api/internal/branch/%d/%s", repoID, branchName)
 	log.GitLogger.Trace("GetProtectedBranchBy: %s", reqURL)
 
-	resp, err := newRequest(reqURL, "GET").SetTLSClientConfig(&tls.Config{
-		InsecureSkipVerify: true,
-	}).Response()
+	resp, err := newInternalReqest(reqURL, "GET").Response()
```
Comment (silverwind) [inline modules/private/branch.go]: Typo `newInternalReqest`

Comment (lafriks) [reply modules/private/branch.go]: @silverwind fixed typo

**Comment by lunny** [comment]
LGTM, but integration test is better.

**Comment by lafriks** [comment]
Can not create integration test for this as it requires external ssh server

**Comment by sapk** [comment]
LGTM

**Review by sapk — APPROVED** [review]

**Pull request merged**